### PR TITLE
Fix FreeBSD build with GCC

### PR DESCRIPTION
**Pull request by arichardson**
And one minor follow-up cleanup.

### CODE REVIEW

File: capsicum-test.cc [hunk 1]
```diff
@@ -72,11 +72,14 @@ char ProcessState(int pid) {
   }
   unsigned int count = 0;
   struct procstat *prstat = procstat_open_sysctl();
-  EXPECT_NE(NULL, prstat) << "procstat_open_sysctl failed.";
+  EXPECT_NE(nullptr, prstat) << "procstat_open_sysctl failed.";
   errno = 0;
   struct kinfo_proc *p = procstat_getprocs(prstat, KERN_PROC_PID, pid, &count);
   if (p == NULL || count == 0) {
```
Comment (jrtc27) [inline capsicum-test.cc:78]: How come this one isn't an issue?

Comment (arichardson) [reply capsicum-test.cc:78]: I think GCC lets you compare to zero/__null, but template instantiations use unsigned long?

File: makefile [hunk 2]
```diff
@@ -18,7 +18,11 @@ mini-me: mini-me.c
 mini-me.noexec: mini-me
 	cp mini-me $@ && chmod -x $@
 mini-me.setuid: mini-me
-	rm -f $@ && cp mini-me $@&& sudo chown root $@ && sudo chmod u+s $@
+ifneq ($(NO_ROOT),)
```
Comment (daviddrysdale) [inline makefile]: What's `NO_ROOT` ?

Comment (arichardson) [reply makefile]: That's a variable you have to set on the command line. I can drop this commit, but it was useful when testing on a shared server where I don't have root access.

File: capsicum-test.cc (hunk 1 above)
Comment (arichardson) [reply capsicum-test.cc:78]: Or more likely: literal zero is fine, but by the time the template is instatiated this is a pointer-integer comparison.

File: makefile (hunk 2 above)
Comment (jrtc27) [reply makefile]: I think in that situation it makes sense to just disable mini-me.setuid entirely, not create one that's wrong.

Comment (arichardson) [reply makefile]: I'll drop that change and make a new one that can deal with missing mini-me.setuid.

File: capsicum-test.cc (hunk 1 above)
Comment (jrtc27) [reply capsicum-test.cc:78]: Ah right, testing::internal::CmpHelperNE. Blegh

**Review by daviddrysdale — APPROVED** [review]

**Pull request merged**